### ISSUE TITLE
Machine 0.5.0 fails to create with virtualbox driver on Win7

### ISSUE THREAD

**Issue by kpalf** [docker/machine#2318]
Using Toolbox 1.9 on Win7 Pro.

Launching the quickstart terminal fails. (I've added a -D to get extra info).  Tried hunting bug reports and forums but not seeing any reports.  I have a colleague with same issue on another laptop.

Creating Machine default...
Docker Machine Version: 0.5.0 (04cfa58)
Found binary path at .\docker-machine-driver-virtualbox.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:51174
Error loading driver "virtualbox": dial-http tcp 127.0.0.1:51174: read tcp 127.0.0.1:51175->127.0.0.1:51174: wsarecv: An existing connection was forcibly closed by the remote host.
Looks like something went wrong... Press any key to continue...

Can occasionally see different output, but same result:
$ docker-machine -D create -d virtualbox default
Docker Machine Version: 0.5.0 (04cfa58)
Found binary path at C:\Program Files\Docker Toolbox\docker-machine-driver-virtualbox.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:61709
() Calling RpcServerDriver.GetVersion
Using API Version 1
() Calling RpcServerDriver.SetConfigRaw
() Calling RpcServerDriver.GetMachineName
(flag-lookup) Calling RpcServerDriver.GetCreateFlags
Making call to close connection to plugin binary
Making call to close driver server
(flag-lookup) Calling RpcServerDriver.Close
Successfully made call to close driver server
Found binary path at C:\Program Files\Docker Toolbox\docker-machine-driver-virtualbox.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:61713
Error loading driver "virtualbox": dial-http tcp 127.0.0.1:61713: read tcp 127.0.0.1:61714->127.0.0.1:61713: wsarecv: An existing connection was forcibly closed by the remote host.

**Comment by nathanleclaire** [comment]
Any unusual networking configuration on your computer?  Proxy?  VPN?  Firewall?  Antivirus?

**Comment by kpalf** [comment]
Apologies Nathan, I should have given more info. DockerCon != Sleep, but then from the time of your response I guess you are the same. ;-)
This is a corporate laptop image, so yes I have "CheckPoint Endpoint Security E75.10" for VPN, and "McAfee VirusScan Enterprise and AntiSpyware Enterprise 8.8" for AV. Normal windows firewall, enabled. Pretty typical corporate setup I expect.
The other colleague I mentioned with same error message was trying on a home laptop and that apparently has Avast for AV, and Barracuda VPN client.
Previously I'd been using Boot2Docker 1.7.1, but that is completely uninstalled.

**Comment by kpalf** [comment]
Nathan or anyone - any update or progress on this issue?
Thanks, Kev

**Comment by nathanleclaire** [comment]
Kev, none so far, but I'll keep you posted if I find anything out.

**Comment by davidkarlsen** [comment]
Checkpoint here to - for some odd reason it adds the routes to the docker-machine over the VPN - so you need to do a:
```
route delete 192.168.99.0 mask 255.255.255.0 <yourVpnGw>
```

then I got it working even when on VPN (which is a requirement for me - so that I can pull the remote repos over VPN).

**Comment by joshuacalloway** [comment]
Try starting VirtualBox first then run docker toolbox 
Also remove the default machine ( docker-machine.exe remove default )

**Comment by kpalf** [comment]
I can confirm the issue is still there with machine 0.5.4 (from toolbox 1.9.1f).  
In response to above, I don't see any odd routes when the VPN is not running (so none need deleting); and makes no difference if VirtualBox is started first.  Seems like an issue between docker-machine and the driver, not between the driver and vbox, but I could be wrong.  
Seems similar to toolbox issue #340.  Happy to try a debug/diagnostic build.

**Comment by djsylvester** [comment]
I had similar issues. Although I use Avast anit-virus and Outpost Firewall Pro (free version doesn't cause this) it may help you troubleshoot. https://github.com/docker/toolbox/issues/272

**Comment by pietv** [comment]
Any plans to fix this issue soon? It's still present in 0.5.5. Running Docker Machine on AppVeyor.

**Comment by nathanleclaire** [comment]
@pietv We are working on it but it's tricky to reproduce.  Do you have it reproducing reliably?

**Comment by compleatang** [comment]
@nathanleclaire Yes. With 0.5.5 it was happening every time our tests ran (I work with @pietv so we're talking about the same thing). Here's a sample build result -> https://ci.appveyor.com/project/eris-ltd/eris-cli/build/1.0.364-winConFix . We're working around the issue by using 0.4.1 which was before the move to plugin model and that works reliably. 

The build system uses: Windows Server 2012 R2. 

I was also able to reproduce when I was trying to debug something else by setting up a Windows Enterprise box on Azure and using remote desktop, `choco install docker-machine` and it happened numerous times until I downgraded to 0.4.1.

**Comment by nathanleclaire** [comment]
@compleatang Any chance I could get that same CI output with `--debug` enabled?

**Comment by compleatang** [comment]
Sure. Let me change it and push a build now.

**Comment by compleatang** [comment]
Build is queued, but it'll be here -> https://ci.appveyor.com/project/eris-ltd/eris-cli/build/1.0.440-winTest

**Comment by compleatang** [comment]
Debug output here -> https://ci.appveyor.com/project/eris-ltd/eris-cli/build/1.0.441-winTest

There was a panic, so am assuming that will be helpful for debugging purposes. Thanks for all your work, we're heavy docker-machine users and very thankful for your efforts.

**Comment by nathanleclaire** [comment]
Yes, it's very helpful.  Actually, handling panics in the plugin / server side is something that I had originally intended to work on implementing but it slipped through the cracks.  I wonder if this is what's behind all of the "connection reset by errors" and if we're just now getting all the logs thanks to @dgageot efforts.

> Thanks for all your work, we're heavy docker-machine users and very thankful for your efforts.

Nice and very appreciated!

**Comment by nathanleclaire** [comment]
Filed https://github.com/docker/machine/pull/2823 which should hopefully at least resolve the panics that @compleatang.  It won't fix the bug that's causing an out of bounds error though, that seems to be a bug in the AWS driver.

**Comment by nathanleclaire** [comment]
@compleatang Let me know if you're interested in trying a master build to see if it fixes the issue.  I have a feeling the #2823 I submitted will serve as a workaround for the panic but there's a deeper underlying issue (some kind of invalid state / input) at play.

**Comment by compleatang** [comment]
will give it a go over the weekend. it was on our windows CI server, but doesn't look too bad to build. will revert with results.

**Comment by kpalf** [comment]
I've been trying to collect a series of different logs for you.
We've not had working Docker on Windows (with VirtualBox) now since 1.9.0 (just before DockerCon EU), which was when I originally filed this defect, and the focus was on this specific error message.   I realise the releases have moved on since then, but I still get 100% failure but it feels like maybe a race as to weather I get the "wsarecv" error, or maybe the more recent releases are better at not closing the connection with the driver.
On issue #2823 that seemed it might be related I offered to test the new 0.6.0-dev binary that was offered but I think the logs I'm providing are really better as a continuation of the defect I originally reported.
This is my work laptop but its been crazy busy at work so this is mostly squeezing in on evening / weekend between family duties. So, apologies for the delay.

I can't get GitHub to accept the zip files of logs as an attachment.  I have temp uploaded to OneDrive - let me know when you've grabbed them.
   http://1drv.ms/1Ozmnmz 

I've been doing a full ToolBox uninstall (followed by Git uninstall, and VirtualBox uninstall, and deleting ~/.docker/ and ~/.VirtualBox) for each version tried.

Look like I got the rates for types of failures incorrect.  I have had 100% failure rate on 1.9.0 and 1.9.1 (various suffix), but when I was trying on Saturday it looks like only around 5-10% reported the specific "wsarecv: An existing connection was forcibly closed by the remote host" error.  But I must stress that in ALL cases it failed to start, even then the end of the debug logs says "VM default has been successfully started".

I've collected a collection of logs for you.  From 1.9.0c (0.5.0), 1.9.1e (0.5.3), 1.9.1h, and 1.9.1h with Special binary you supplied.

I've been editing the start.sh script in "Program Files/Docker Toolbox" to change the line to create the machine as follows to collect debug output:
  --  $DOCKER_MACHINE create -d virtualbox $VM
  ++  $DOCKER_MACHINE -D create -d virtualbox $VM > ~/output.txt 2>&1

With Machine 0.5.0 in Toolbox 1.9.0c, that didn't really give us much if any output.  It did seem to reliably fail with the above error, but as there was no debug I moved on.  One in the attached logs anyway.

I ran quite a few cycles with Toolbox 1.9.1e, initially without network while I was out on Sat. Failed every time. One in the attached logs. Noticed I wasn't always getting the above error but still 100% failure rate.  Apologies I wasn't deleting the folder in the home directory each time I ran start.sh.
I then ran some more once home with network.  Again, just tried repeatedly running start.sh and saw the specific error was between 5-10%, but 100% failure.  I do have a set of logs attached where that specific error occurred.  You'll also see the vbox logs from a bunch of previous runs.
Most often we just see "Looks like something went wrong... Press any key to continue...".  Occasionally we get a few more lines.

Next on to trying 1.9.1h.  (1.9.1 definitely seems much slower to get to failure)
Again first run after clean install with debugging captured and attached.  Failed as usual, but not with the specific "wsarecv" error.
Second run failed but with an extra line in console of "exit status 1", and I didn't get my debug output file - looks like it didn't run the create command.  Looks like the start script changed since prior versions.
Started VirtualBox and removed the VM (+ deleted files), and quit VirtualBox again.
Tried start.sh again.  Out similar to first 1.9.1h run - same failure mode.  The changed start.sh means we can't just keep running it to try to get to the error we wanted.  Didn't try to track down which command was failing.

Next to try with the test binary.
Started VirtualBox and removed the VM (+ deleted files), and quit VirtualBox again.
Also deleted ~/.docker/ and ~/.VirtualBox.  So next run will recreate and re-download the iso.
Replaced machine binary with that supplied by Nathan 0.6.0dev-build4291b59
Pretty similar failure mode to with machine 0.5.5.
Apologies, I don't seem to be provoking the "wsarecv" error.  

I compared the debug output of the 0.5.3 run that did give the "wsarecv" error vs one of the 0.5.5 runs that failed but without the error.
There are various differences as you can see for yourself.  Looked like the failing one had these extra 4 lines near the beginning:
  (flag-lookup) Calling .GetCreateFlags
  ++ Making call to close connection to plugin binary
  ++ Making call to close driver server
  ++ (flag-lookup) Calling .Close
  ++ Successfully made call to close driver server
  Found binary path at C:\Program Files\Docker Toolbox\docker-machine.exe

So I'm not sure I'm helping you with the "wsarecv" error, but maybe some of my logs help find/fix the other failure modes.  As a dev manager I realise this is not clean for helping you separate 
defects, but I'll let you decide how you want to partition them.  Hope the logs are useful.

Kev

**Comment by kpalf** [comment]
For completeness, I also just downloaded and tried Toolbox 1.9.1i (machine 0.5.6).  Same outcome, though interestingly the debug output has sections in different order to 0.5.5, but the 0.5.6 release notes indicated changes in those areas of the virtualbox driver so I guess expected.  Still failed though.

**Comment by achekulaev** [comment]
In my VmWare I have Windows 7 snapshot where I can repeat this bug every time.
It fails first time. I delete it with `docker-machine rm` and next time it creates fine.

```
docker-machine -D create --driver=virtualbox --virtualbox-disk-size 20000 --virtualbox-memory 1024 --virtualbox-hostonly-cidr 192.168.64.1/24 --virtualbox-no-share docksal
Docker Machine Version:  0.8.2, build e18a919
Found binary path at C:\Users\alex\.babun\cygwin\home\alex\.docksal\bin\docker-machine.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:49260
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Found binary path at C:\Users\alex\.babun\cygwin\home\alex\.docksal\bin\docker-machine.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:49264
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(docksal) Calling .GetMachineName
(docksal) Calling .DriverName
(docksal) Calling .GetCreateFlags
(docksal) Calling .SetConfigFromFlags
Creating CA: C:\Users\alex\.docker\machine\certs\ca.pem
Creating client certificate: C:\Users\alex\.docker\machine\certs\cert.pem
Running pre-create checks...
(docksal) Calling .PreCreateCheck
(docksal) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe --version
(docksal) DBG | STDOUT:
(docksal) DBG | {
(docksal) DBG | 5.1.2r108956
(docksal) DBG | }
(docksal) DBG | STDERR:
(docksal) DBG | {
(docksal) DBG | }
(docksal) DBG | Hyper-V is not installed.
(docksal) DBG | %!(EXTRA *exec.Error=exec: "vmms.exe": executable file not found in %PATH%)COMMAND: wmic cpu get VirtualizationFirmwareEnabled
(docksal) DBG | Couldn't check that VT-X/AMD-v is enabled. Will check that the vm is properly created: exit status 2147749911
(docksal) Image cache directory does not exist, creating it at C:\Users\alex\.docker\machine\cache...
(docksal) No default Boot2Docker ISO found locally, downloading the latest release...
(docksal) Latest release for github.com/boot2docker/boot2docker is v1.12.3
(docksal) Downloading C:\Users\alex\.docker\machine\cache\boot2docker.iso from https://github.com/boot2docker/boot2docker/releases/download/v1.12.3/boot2docker.iso...
(docksal) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(docksal) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe list hostonlyifs
(docksal) DBG | STDOUT:
(docksal) DBG | {
(docksal) DBG | Name:            VirtualBox Host-Only Ethernet Adapter
(docksal) DBG | GUID:            361ad37e-d162-4f8e-873b-7e50605a4791
(docksal) DBG | DHCP:            Disabled
(docksal) DBG | IPAddress:       192.168.56.1
(docksal) DBG | NetworkMask:     255.255.255.0
(docksal) DBG | IPV6Address:     fe80:0000:0000:0000:9440:afe8:3fd4:e7b9
(docksal) DBG | IPV6NetworkMaskPrefixLength: 64
(docksal) DBG | HardwareAddress: 0a:00:27:00:00:16
(docksal) DBG | MediumType:      Ethernet
(docksal) DBG | Status:          Up
(docksal) DBG | VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter
(docksal) DBG |
(docksal) DBG | }
(docksal) DBG | STDERR:
(docksal) DBG | {
(docksal) DBG | }
(docksal) Calling .GetConfigRaw
Creating machine...
(docksal) Calling .Create
(docksal) DBG | local Boot2Docker ISO version:  v1.12.3
(docksal) Copying C:\Users\alex\.docker\machine\cache\boot2docker.iso to C:\Users\alex\.docker\machine\machines\docksal\boot2docker.iso...
(docksal) Creating VirtualBox VM...
(docksal) Creating SSH key...
(docksal) DBG | Creating disk image...
(docksal) DBG | Creating 20000 MB hard disk image...
(docksal) DBG | Writing magic tar header
(docksal) DBG | Writing SSH key tar header
(docksal) DBG | Calling inner createDiskImage
(docksal) DBG | &{C:\Program Files\Oracle\VirtualBox\VBoxManage.exe [C:\Program Files\Oracle\VirtualBox\VBoxManage.exe convertfromraw stdin C:\Users\alex\.docker\machine\machines\docksal\disk.vmdk 20971520000 --format VMDK] []  <nil> <nil> <nil> [] <nil> <nil> <nil> <nil> <nil> false [] [] [] [] <nil> <nil>}
(docksal) DBG | Starting command
(docksal) DBG | Copying to stdin
(docksal) DBG | Filling zeroes
(docksal) DBG | Closing STDIN
(docksal) DBG | Waiting on cmd
(docksal) DBG | Converting from raw image file="stdin" to file="C:\Users\alex\.docker\machine\machines\docksal\disk.vmdk"...
(docksal) DBG | Creating dynamic image with size 20971520000 bytes (20000MB)...
Wrapper Docker Machine process exiting due to closed plugin server (read tcp 127.0.0.1:49265->127.0.0.1:49264: wsarecv: An existing connection was forcibly closed by the remote host.)
Error creating machine: Error in driver during machine creation: read tcp 127.0.0.1:49265->127.0.0.1:49264: wsarecv: An existing connection was forcibly closed by the remote host.
open C:\Users\alex\.docker\machine\machines\docksal\docksal\Logs\VBox.log: The system cannot find the path specified.
notifying bugsnag: [Error creating machine: Error in driver during machine creation: read tcp 127.0.0.1:49265->127.0.0.1:49264: wsarecv: An existing connection was forcibly closed by the remote host.]
```

**Comment by budhrg** [comment]
Not sure about Window 7 but  in Windows 10 with latest stable release `docker-machine-Windows-x86_64.exe version 0.8.2, build e18a919` it is working fine for me.

```
PS D:\> docker-machine-Windows-x86_64.exe version
docker-machine-Windows-x86_64.exe version 0.8.2, build e18a919

PS D:\> docker-machine-Windows-x86_64.exe create -d virtualbox default
Running pre-create checks...
(default) No default Boot2Docker ISO found locally, downloading the latest release...
(default) Latest release for github.com/boot2docker/boot2docker is v1.12.3
(default) Downloading C:\Users\budhram\.docker\machine\cache\boot2docker.iso from https://github.com/boot2docker/boot2docker/releases/download/v1.12.3/boot2do
cker.iso...
(default) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
Creating machine...
(default) Copying C:\Users\budhram\.docker\machine\cache\boot2docker.iso to C:\Users\budhram\.docker\machine\machines\default\boot2docker.iso...
(default) Creating VirtualBox VM...
(default) Creating SSH key...
(default) Starting the VM...
(default) Check network to re-create if needed...
(default) Windows might ask for the permission to create a network adapter. Sometimes, such confirmation window is minimized in the taskbar.
(default) Found a new host-only adapter: "VirtualBox Host-Only Ethernet Adapter #3"
(default) Windows might ask for the permission to configure a network adapter. Sometimes, such confirmation window is minimized in the taskbar.
(default) Windows might ask for the permission to configure a dhcp server. Sometimes, such confirmation window is minimized in the taskbar.
(default) Waiting for an IP...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with boot2docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: D:\my_bin\docker-machine-Windows-x86_64.exe env default

PS D:\> docker-machine-Windows-x86_64.exe env
$Env:DOCKER_TLS_VERIFY = "1"
$Env:DOCKER_HOST = "tcp://192.168.99.100:2376"
$Env:DOCKER_CERT_PATH = "C:\Users\budhram\.docker\machine\machines\default"
$Env:DOCKER_MACHINE_NAME = "default"
# Run this command to configure your shell:
# & "D:\my_bin\docker-machine-Windows-x86_64.exe" env | Invoke-Expression
```

**Comment by achekulaev** [comment]
Windows 10 works fine for me too. I only meet this bug in Windows 7.

**Comment by jcollum-cambia** [comment]
I'm seeing it in Win 10:

```
Command failed: C:\Program Files\Docker Toolbox\docker-machine.exe -D create -d virtualbox --virtualbox-memory 2048 default,

Docker Machine Version: 0.11.0,
 
build 5b27455,

Found binary path at C:\Program Files\Docker Toolbox\docker-machine.exe,

Launching plugin server for driver virtualbox,

Plugin server listening at address 127.0.0.1:50693,
() Calling .GetVersion,
Using API Version 1,
() Calling .SetConfigRaw,
() Calling .GetMachineName,
(flag-lookup) Calling .GetMachineName,
(flag-lookup) Calling .DriverName,
(flag-lookup) Calling .GetCreateFlags,
Found binary path at C:\Program Files\Docker Toolbox\docker-machine.exe,
Launching plugin server for driver virtualbox,
Plugin server listening at address 127.0.0.1:50697,
() Calling .GetVersion,
Using API Version 1,
() Calling .SetConfigRaw,
() Calling .GetMachineName,
(default) Calling .GetMachineName,
(default) Calling .DriverName,
(default) Calling .GetCreateFlags,
(default) Calling .SetConfigFromFlags,
(default) Calling .PreCreateCheck,
(default) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe --version,
(default) DBG | STDOUT:,
(default) DBG | {,
(default) DBG | 5.1.24r117012,
(default) DBG | },
(default) DBG | STDERR:,
(default) DBG | {,
(default) DBG | },
(default) DBG | Hyper-V is not installed.,
(default) DBG | %!(EXTRA *exec.Error=exec: "vmms.exe": executable file not found in %PATH%)COMMAND: wmic cpu get VirtualizationFirmwareEnabled,
(default) DBG | STDOUT:,
(default) DBG | {,
(default) DBG | VirtualizationFirmwareEnabled ,
(default) DBG | TRUE ,
(default) DBG | ,
(default) DBG | },
Error with pre-create check: "Get https://api.github.com/repos/boot2docker/boot2docker/releases/latest: read tcp 10.26.178.67:50699->192.30.255.117:443: wsarecv: An existing connection was forcibly closed by the remote host.",
open C:\Users\xxxxxxx\.docker\machine\machines\default\default\Logs\VBox.log: The system cannot find the path specified.,
notifying bugsnag: [Error with pre-create check: "Get https://api.github.com/repos/boot2docker/boot2docker/releases/latest: read tcp 10.26.178.67:50699->192.30.255.117:443: wsarecv: An existing connection was forcibly closed by the remote host."],
```

Windows 10.0.14393

I'm on a corporate network behind a proxy. Not using a VPN at this time.

**Comment by gtseres** [comment]
I am seeing it in:
```
OS: Windows 10
Virtualbox Version: 5.1.26 r117224 (Qt5.6.2)
Docker machine version: 0.12.2, build 9371605
```

It's a intermittent issue though, as for example for two subsequent runs of the `docker-machine ssh` command, the first one failed and the second one succeeded. Not sure if this has happened with other commands though.

**Comment by gitGksgk** [comment]
Not sure if it has been mentioned, but installing vc redistributable package fixes the issue for me on Windows. I try this because running a local php server also failes for a lack of some dll.. Maybe it's some sort of socket problem..

**Comment by almgwary** [comment]
> Not sure if it has been mentioned, but installing vc redistributable package fixes the issue for me on Windows. I try this because running a local php server also failes for a lack of some dll.. Maybe it's some sort of socket problem..

 
@gitGksgk Your solution worked for me, Thank you

**Comment by JoseFMP** [comment]
Which version of VC redistributable solves the issue?

**Comment by bitixel** [comment]
Setting up a proxy fixes the issue for me on Windows.

**Comment by gufranpathan** [comment]
I'm using Windows 10 and i can't pull any image when I "Switch to Windows Containers".

**Comment by bilalmalik777** [comment]
I am using window 10 with 1809 build version, i am unable to pull any image, when i switch to windows container., neither on my local machine nor remote server. But pulling image command work correctly when i switch to linux container. If you need more information let me know

**Comment by usego** [comment]
Same story with Winserver 2019 running at local hosting facility (but no issues in AWS)

`
docker pull mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019
windowsservercore-ltsc2019: Pulling from windows/servercore/iis
65014b3c3121: Downloading [==================================================>]  1.521GB/1.521GB
0e787b79aad1: Downloading [==================================================>]  412.4MB/412.4MB
da1f742f6223: Download complete
5004cdfbdb1c: Download complete
b9bd0eb91d37: Download complete
read tcp 10.100.0.12:52570->204.79.197.219:443: wsarecv: An existing connection was forcibly closed by the remote host.
`
